File: 2025-06-June-GFG-POTD/June 24 - Lexicographically Largest String After K Deletions/Solution.py
class Solution:
    def maxSubseq(self, s: str, k: int) -> str:
        res = []
        for char in s:
            while k > 0 and res and res[-1] < char:
                res.pop()
                k -= 1
            res.append(char)
        if k > 0:
            res = res[:-k]
        return ''.join(res)

File: 2025-06-June-GFG-POTD/June 24 - Lexicographically Largest String After K Deletions/Solution.cpp
#include <string>
using namespace std;

class Solution {
public:
    string maxSubseq(string s, int k) {
        string res;
        for (char c : s) {
            while (k > 0 && !res.empty() && res.back() < c) {
                res.pop_back();
                k--;
            }
            res.push_back(c);
        }
        if (k > 0) {
            res = res.substr(0, res.size() - k);
        }
        return res;
    }
};
